hoomd/SnapshotSystemData.cc

```diff
@@ -39,6 +39,23 @@ void SnapshotSystemData<Real>::replicate(unsigned int nx, unsigned int ny, unsig
     pair_data.replicate(n, old_n);
     }
 
+template<class Real> void SnapshotSystemData<Real>::wrap()
+    {
+    vec3<double> new_pos;
+    vec3<double> old_pos;
+    for (unsigned int i = 0; i < particle_data.size; i++)
+        {
+        do
+        {
+        new_pos = particle_data.pos[i];
+        old_pos = new_pos;
+        global_box.wrap(new_pos, particle_data.image[i]);
+        particle_data.pos[i] = new_pos;
+        }
+        while (new_pos != old_pos);
+        }
+    }
+
```

Comment:
> re-implementing the lower level box wrap method
Perhaps instead this is where we could make a method `BoxDim.wrap_multiple` so that `BoxDim.wrap` stays fast for single wraps only?

Reply:
We could mimic freud's implementation with a call to `makeFraction`, modding by 1, then `makeCoordinates` and updating the image by static_cast<int>(fraction).